rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1047,7 +1069,7 @@ stream {
 
             {{ $proxySetHeader }} X-Request-ID           $req_id;
             {{ $proxySetHeader }} X-Real-IP              $the_real_ip;
-            {{ if $all.Cfg.ComputeFullForwardedFor }}
+            {{ if and $all.Cfg.UseForwardedHeaders $all.Cfg.ComputeFullForwardedFor }}
```

Comment:
`$full_x_forwarded_for` is the same as `$proxy_add_x_forwarded_for` but done "by hand" in a map above.

Maybe this is so it appends the proxy protocol addr instead of `$remote_addr` if using proxy protocol, and nginx's `$proxy_add_x_forwarded_for` doesn't?

Reply:
yeah that makes sense @Dirbaio 

Why do you require `$all.Cfg.UseForwardedHeaders` to be `true` here though?

Reply:
`ComputeFullForwardedFor` has no effect when `UseForwardedHeaders = false`. 

There are 3 cases to consider:

- `UseForwardedHeaders = false`: incoming X-Forwarded-For value is discarded and replaced with the source IP nginx-ingress sees.
- `UseForwardedHeaders = true, ComputeFullForwardedFor = false`: incoming X-Forwarded-For value is passed through if present, otherwise it's filled with the source IP nginx-ingress sees.
- `UseForwardedHeaders = true, ComputeFullForwardedFor = true`: incoming X-Forwarded-For value is passed through, appended with the source IP nginx-ingress sees.

For case 3 we use `$full_x_forwarded_for`, which has been computed higher in the template. 
For cases 1, 2 we use `$the_real_ip`, which higher in the template has been set to be the appropriate source IP depending on `UseForwardedHeaders`